### PR TITLE
Refactor the New Notes Notification code

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
#1386

## Description
Refactored away one of the unnecessary home feed predicates. Also removed the RefreshController protocol, which seemed to be causing SwiftUI problems, and simplified it even more.

## How to test
This is a refactor, so everything should still work as it used to. Specifically, though, look at the home feed and the profile screens.

1. Navigate to Feed
2. Wait for new notes to appear (or post one from another device)
3. Tap "New notes available"
4. Ensure the view is refreshed and scrolls to the top
5. Scroll down in the Feed
6. Tap a note to see details
7. Tap back
8. Ensure the Feed still appears in the same place as it was after you scrolled
9. Navigate to a profile
10. Scroll, tap, go back, refresh, make sure it all works.

### CODE REVIEW

File: Nos/Controller/RefreshController.swift [hunk 1]
```diff
@@ -1,36 +1,19 @@
...
-    /// Update the state of `shouldRefresh` to the given value.
-    func setShouldRefresh(_: Bool)
-
-    /// Updates the last refresh date to the given value.
-    func setLastRefreshDate(_: Date)
-}
-
 /// The default implementation of `RefreshController`.
-@Observable @MainActor class DefaultRefreshController: RefreshController {
+@Observable @MainActor class RefreshController {
```
Comment (martindsq) [inline Nos/Controller/RefreshController.swift:4]: The comment above is now outdated

File: Nos/Models/CoreData/Event+CoreDataClass.swift [hunk 2]
```diff
@@ -325,42 +325,44 @@ public class Event: NosManagedObject, VerifiableEvent {
...
         // swiftlint:disable:next line_length
-        var queryString = "((kind = 1 AND SUBQUERY(eventReferences, $reference, $reference.marker = 'root' OR $reference.marker = 'reply' OR $reference.marker = nil).@count = 0) OR kind = 6 OR kind = 30023) AND author.muted = 0 AND createdAt <= %@ AND deletedOn.@count = 0"
-        var arguments: [CVarArg] = [before as CVarArg]
-        if let relay {
-            queryString.append(" AND ANY seenOnRelays = %@")
-            arguments.append(relay)
-        } else {
-            queryString.append(" AND (ANY author.followers.source = %@ OR author = %@)")
-            arguments += [user, user]
+        var queryString = "((kind = 1 AND SUBQUERY(eventReferences, $reference, $reference.marker = 'root' OR $reference.marker = 'reply' OR $reference.marker = nil).@count = 0) OR kind = 6 OR kind = 30023) AND author.muted = 0 AND deletedOn.@count = 0"
```
Comment (martindsq) [inline Nos/Models/CoreData/Event+CoreDataClass.swift:342]: How about refactoring this function with a NSCompoundPredicate. I think the code will look much better and will be easier to modify. We are already using it in other places.

File: Nos/Controller/RefreshController.swift [hunk 3]
```diff
@@ -1,36 +1,19 @@
...
+    /// Whether a refresh should begin or not.
```
Comment (martindsq) [inline Nos/Controller/RefreshController.swift:5]: I think I need a better description about this property. The main question I have when seeing this property is: why a controller named RefreshController has a property named shouldRefresh, shouldn't it be always refreshing? Or it is more related to enabling or disabling it?

**Review by martindsq — CHANGES_REQUESTED** [review]

File: Nos/Controller/RefreshController.swift (hunk 3 above)
Comment (joshuatbrown) [reply Nos/Controller/RefreshController.swift:5]: I'll rename it to `startRefresh` and update the comment.

Comment (joshuatbrown) [reply Nos/Controller/RefreshController.swift]: Done!

File: Nos/Controller/RefreshController.swift (hunk 1 above)
Comment (joshuatbrown) [reply Nos/Controller/RefreshController.swift:4]: Fixed!

File: Nos/Models/CoreData/Event+CoreDataClass.swift (hunk 2 above)
Comment (joshuatbrown) [reply Nos/Models/CoreData/Event+CoreDataClass.swift:342]: I can do that.

Comment (joshuatbrown) [reply Nos/Models/CoreData/Event+CoreDataClass.swift]: done!

**Review by martindsq — APPROVED** [review]
I really like your refactor using NSCompoundPredicate, it's easier to read.

**Pull request merged**